keras/callbacks.py

```diff
@@ -1318,15 +1327,17 @@ class ModelCheckpoint(Callback):
 
     def __init__(
         self,
-        filepath,
-        monitor="val_loss",
-        verbose=0,
-        save_best_only=False,
-        save_weights_only=False,
-        mode="auto",
-        save_freq="epoch",
-        options=None,
-        initial_value_threshold=None,
+        filepath: Union[str, os.PathLike],
+        monitor: str = "val_loss",
+        verbose: int = 0,
+        save_best_only: bool = False,
+        save_weights_only: bool = False,
+        mode: str = "auto",
+        save_freq: Union[int, str] = "epoch",
+        options: Union[
+            tf.train.CheckpointOptions, tf.saved_model.SaveOptions, None
+        ] = None,
```

Comment:
In general we should avoid explicit types when the expect type is a custom class, like in the above case. It creates significant tech debt and maintenance burden going forward.